core/src/test/scala/unit/kafka/tools/DumpLogSegmentsTest.scala

```diff
@@ -310,6 +334,38 @@ class DumpLogSegmentsTest {
     None
   }
 
+  // Returns the total bytes of the batches specified
+  private def readPartialBatchesBytes(lines: util.ListIterator[String], limit: Int): Int = {
+    val sizePattern: Regex = raw".+?size:\s(\d+).+".r
+    var batchesBytes = 0
+    var batchesCounter = 0
+    while (lines.hasNext) {
+      if (batchesCounter >= limit){
+        return batchesBytes
+      }
+      val line = lines.next()
+      if (line.startsWith("baseOffset")) {
+        line match {
+          case sizePattern(size) => batchesBytes += size.toInt
+          case _ => throw new IllegalStateException(s"Failed to parse and find size value for batch line: $line")
+        }
+        batchesCounter += 1
+      }
+    }
+    return batchesBytes
+  }
+
+  private def countBatches(lines: util.ListIterator[String]): Int = {
+    var countBatches = 0
+    while (lines.hasNext) {
+      val line = lines.next()
+      if (line.startsWith("baseOffset")) {
+        countBatches += 1
+      }
+    }
+    return countBatches
```

Comment:
No need for return.

Reply:
sorry I dont get why the return is not needed here :)

Reply:
I just meant that in scala you can just do `countBatches` instead of `return countBatches` in the last statement.

Reply:
cool, thanks!